subprojects/language-java/src/test/groovy/org/gradle/api/tasks/compile/CompileOptionsTest.groovy

```diff
@@ -16,135 +16,205 @@
 
-    @Before public void setUp()  {
+    def setup()  {
         compileOptions = new CompileOptions()
         compileOptions.debugOptions = [optionMap: {TEST_DEBUG_OPTION_MAP}] as DebugOptions
         compileOptions.forkOptions = [optionMap: {TEST_FORK_OPTION_MAP}] as ForkOptions
     }
 
-    @Test public void testCompileOptions() {
-        assertTrue(compileOptions.debug)
-        assertTrue(compileOptions.failOnError)
-        assertTrue(compileOptions.warnings)
-
-        assertFalse(compileOptions.deprecation)
-        assertFalse(compileOptions.listFiles)
-        assertFalse(compileOptions.verbose)
-        assertFalse(compileOptions.fork)
-
-        assertThat(compileOptions.compilerArgs, isEmpty())
-        assertNull(compileOptions.encoding)
-        assertNull(compileOptions.bootClasspath)
-        assertNull(compileOptions.extensionDirs)
-
-        assertNotNull(compileOptions.forkOptions)
-        assertNotNull(compileOptions.debugOptions)
+    @SuppressWarnings("GrDeprecatedAPIUsage")
+    def "default compile options"() {
+        expect:
+        compileOptions.debug
+        compileOptions.failOnError
+        compileOptions.warnings
+
+        !compileOptions.deprecation
+        !compileOptions.listFiles
+        !compileOptions.verbose
+        !compileOptions.fork
+
+        compileOptions.compilerArgs.empty
+        compileOptions.encoding == null
+        compileOptions.bootClasspath == null
+        compileOptions.bootstrapClasspath == null
+        compileOptions.extensionDirs == null
+
+        compileOptions.forkOptions != null
+        compileOptions.debugOptions != null
     }
 
-    @Test public void testOptionMapForDebugOptions() {
+    def "option map for debug options"() {
         Map optionMap = compileOptions.optionMap()
-        assertEquals(optionMap.subMap(TEST_DEBUG_OPTION_MAP.keySet()), TEST_DEBUG_OPTION_MAP)
-        assertEquals(optionMap.subMap(TEST_FORK_OPTION_MAP.keySet()), TEST_FORK_OPTION_MAP)
+        expect:
+        optionMap.subMap(TEST_DEBUG_OPTION_MAP.keySet()) == TEST_DEBUG_OPTION_MAP
+        optionMap.subMap(TEST_FORK_OPTION_MAP.keySet()) == TEST_FORK_OPTION_MAP
     }
 
-    @Test public void testOptionMapWithNullables() {
-        Map optionMap = compileOptions.optionMap()
-        Map nullables = [
-                encoding: 'encoding',
-                bootClasspath: 'bootClasspath',
-                extensionDirs: 'extdirs'
-        ]
-        nullables.each {String field, String antProperty ->
-            assertFalse(optionMap.keySet().contains(antProperty))
-        }
-
-        nullables.keySet().each {compileOptions."$it" = "${it}Value"}
-        optionMap = compileOptions.optionMap()
-        nullables.each {String field, String antProperty ->
-            assertEquals("${field}Value" as String, optionMap[antProperty])
-        }
+    @Unroll
+    def "option map with nullable #option"() {
+        expect:
+        !compileOptions.optionMap().keySet().contains(option)
```

Comment:
⭕️ Can we get rid of `AbstractOptions.optionMap()` altogether? I suppose we were using it for calling Ant compilers. But I think we already removed all of those from Gradle core, right? At least I didn't find any usages of the method in Gradle core. Note that I don't necessarily suggest we should remove/deprecate `optionMap` as part of this PR.

Reply:
Do you want to create a card for this?

Reply:
I opened an issue: https://github.com/gradle/gradle/issues/3026
I realized that the whole `AbstractOptions` class is now superfluous.